File: JavaScript/duplicate-encoder.js
function duplicateEncode(word) {
    let result = word.toUpperCase();
    console.log(word);
    arr = [];
    for (i of word) {
        arr.push(i);
    }
    for (let i = 0; i < result.length; i++) {
        if ((result.split(result[i]).length - 1) > 1) {
            arr[i] = ")";
        }
        else {
            arr[i] = "(";
        }
    }
    return arr.join("");
}

File: Python/duplicate-encoder.py
def duplicate_encode(word: str):
    word = word.lower()
    result = [x for x in word]
    for i in range(len(word)):
        if word.count(word[i]) > 1:
            result[i] = ")"
        else:
            result[i] = "("
    return ''.join(result)
